ouroboros-network/test/Test/Ouroboros/Network/PeerSelection/RootPeersDNS.hs

```diff
@@ -290,12 +291,17 @@ mockDNSActions dnsMap dnsTimeoutScript dnsLookupDelayScript =
        Just (Left e)  -> return ([e], [])
        Just (Right a) -> return ([], a)
 
+-- TODO: Change this once upgraded to most recent base version 4.16
+data Solo a = Solo a
+  deriving (Show, Eq)
```

Comment:
Why not import it?

Reply:
[Solo](https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-Tuple.html#t:Solo)

Reply:
Only available in base 4.16, would have to upgrade bounds I think, or it is due to how nix is configured.

Reply:
Not with base; it's version is bound to [ghc version](https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/libraries/version-history), we'll need to wait for `9.2`.